Gemfile

```
@@ -1,58 +1,49 @@
-require 'openssl'
-source 'https://rubygems.org'
-gemspec
+# frozen_string_literal: true
 
-if ENV['RAILS_SOURCE']
-  gemspec path: ENV['RAILS_SOURCE']
-else
-  def get_version_from_gemspec
-    gemspec = eval(File.read('activerecord-cockroachdb-adapter.gemspec'))
+source "https://rubygems.org"
 
-    gem_version = gemspec.dependencies.
-      find { |dep| dep.name == 'activerecord' }.
-      requirement.
-      requirements.
-      first.
-      last
+gemspec
 
-    major, minor, tiny, pre = gem_version.segments
 
-    if pre
-      gem_version.to_s
-    else
-      find_latest_matching_version(major, minor)
+module RailsTag
+  class << self
+    def call
+      req = gemspec_requirement
+      "v" + all_activerecord_versions.find { req.satisfied_by?(_1) }.version
     end
-  end
 
-  def find_latest_matching_version(gemspec_major, gemspec_minor)
-    all_activerecord_versions.
-      reject { |version| version["prerelease"] }.
-      map { |version| version["number"].split(".").map(&:to_i) }.
-      find { |major, minor|
-        major == gemspec_major && (minor == gemspec_minor || gemspec_minor.nil?)
-      }.join(".")
-  end
+    def gemspec_requirement
+      File
+        .foreach("activerecord-cockroachdb-adapter.gemspec", chomp: true)
+        .find { _1[/add_dependency\s.activerecord.,\s.(.*)./] }
+
+      Gem::Requirement.new(Regexp.last_match(1))
```

Another option is to go as we did in the postgis adapter:

```suggestion
      gs = Bundler.load_gemspec("activerecord-postgis-adapter.gemspec")
      gs.dependencies.find { |d| d.name == "activerecord" }
```

and replace `satisfied_by?` with `match?` in the `.call` method. I prefer the current version though as we don't need to load gemspec for such little, and this regexp should be robust enough to last until the decay of ruby anyway